Analytical_Engine/Compute/ExtrudeToVolume.cs

```diff
@@ -0,0 +1,104 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using BH.Engine.Geometry;
+using BH.oM.Geometry;
+
+using BH.oM.Base.Attributes;
+using System.ComponentModel;
+
+using BH.oM.Analytical.Elements;
+
+namespace BH.Engine.Analytical
+{
+    public static partial class Compute
+    {
+        [Description("Takes a region with a Floor Perimeter and creates a collection of polylines which represent the closed volume of the region.")]
+        [Input("region", "A region with a floor perimeter to extrude into a collection of panels.")]
+        [Input("height", "The height of the region, as a double, to calculate the ceiling level of the region. This will be used as the Z value of the perimeter + the given height.")]
+        [Input("tolerance", "The degree of tolerance on the angle calculation for collapsing the regions perimeter to a polyline. Default is equal to BH.oM.Geometry.Tolerance.Angle.")]
+        [Output("polylines", "A collection of polylines which represent the closed volume of the region.")]
+        [PreviousInputNames("region", "room, space")]
+        public static List<Polyline> ExtrudeToVolume(this IRegion region, double height, double tolerance = BH.oM.Geometry.Tolerance.Angle)
+        {
+            if (region == null)
+                return new List<Polyline>();
```

Comment:
@FraserGreenroyd do you know if we landed with a consistent convention around such cases? It tempts me to ask for a change to `return null`, so that `null` provided yields `null` in return

Reply:
Based on our [cheat sheet](https://bhom.xyz/documentation/Development/Concepts/Null-Handling/#cheat-sheet) either an empty list or a `null` is acceptable for this scenario. Given the propensity of issues by returning `null` from the `ExtrudeToVolume` I think previously it was decided to return an empty list when the method existed within the Environment Engine for further calculations which is probably why it's doing the same here now.